onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py

```diff
@@ -0,0 +1,576 @@
+    onnx_model = onnx.load_model(temp_path, load_external_data=True)
+    save_onnx_model(
+        onnx_model,
+        output_path,
+        f"{args.model_name}_decoder_with_past_model_fp32.onnx.data",
+    )
+    del onnx_model
+    temp_dir.cleanup()
+
+    logger.info(f"The {args.model_name} ONNX model has been successfully created with the TorchScript exporter!")
+
+
+def remove_existing_files(output_path: str):
+    for filename in os.listdir(output_path):
+        filepath = os.path.join(output_path, filename)
+        if ".onnx" in filename or ".onnx.data" in filename:
+            os.remove(filepath)
+            logger.warning(f"Removing {filepath}")
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-m",
+        "--model_name",
+        required=True,
+        help="Model name in Hugging Face",
+    )
+
+    parser.add_argument(
+        "-i",
+        "--input",
+        required=False,
+        default=os.path.join("."),
+        help="Directory path to PyTorch model and associated files if saved on disk",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        required=False,
+        default=os.path.join(".", "llama_onnx_models"),
+        help="Directory path to save exported model files in",
+    )
+
+    parser.add_argument(
+        "-p",
+        "--precision",
+        required=False,
+        type=Precision,
+        default=Precision.FLOAT32,
+        choices=[Precision.FLOAT32, Precision.FLOAT16, Precision.INT8],
+        help="Precision to export model in",
+    )
+
+    parser.add_argument(
+        "-e",
+        "--execution_provider",
+        required=False,
+        default="cpu",
+        choices=["cpu", "cuda", "rocm"],
+        help="Execution provider to verify parity with",
+    )
+
+    parser.add_argument(
+        "-q",
+        "--quantization_method",
+        default="",
+        choices=["smooth_quant", "quantize_dynamic"],
+        help="Run a specific quantization algorithm. Need to install extra packages in `requirements-quant.txt` for SmoothQuant.",
+    )
+
+    smooth_quant_group = parser.add_argument_group("smooth_quant")
+
+    smooth_quant_group.add_argument(
+        "--smooth_quant_alpha",
+        required=False,
+        default=0.8,
+        type=float,
+        help="Strength to control migration difficulty from activation to weights. Default is 0.8 to match value \
+              used in original paper for LLaMA. Paper recommends using values in [0.4, 0.6] range. \
+              Link to paper: https://arxiv.org/pdf/2211.10438.pdf",
+    )
+
+    smooth_quant_group.add_argument(
+        "--smooth_quant_dataset",
+        required=False,
+        default="NeelNanda/pile-10k",
+        help="Path to dataset for calibration during quantization",
+    )
+
+    smooth_quant_group.add_argument(
+        "--pad_max",
+        required=False,
+        default=196,
```

Comment:
why the default value is 196?

Reply:
The SmoothQuant implementation is based on [this example](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/quantization/language_model/llama/main.py) and the default value for `pad_max` is used [here](https://github.com/microsoft/onnxruntime-inference-examples/blob/07c4d037541f74327b8bf10a162db6c4e9bfdbd7/quantization/language_model/llama/main.py#L90-L94).